src/plugins/mer/mersettings.cpp

```diff
@@ -59,6 +60,30 @@ MerSettings::MerSettings(QObject *parent)
     Q_ASSERT(s_instance == 0);
     s_instance = this;
 
+    // After SDK update device model name collisions might occur
+    const QMap<QString, MerEmulatorDeviceModel> deviceModels = deviceModelsRead(globalDeviceModelsFileName());
+    QMap<QString, MerEmulatorDeviceModel> userDeviceModels = deviceModelsRead(deviceModelsFileName());
+    const QSet<QString> collisions = deviceModels.keys().toSet()
+            .intersect(userDeviceModels.keys().toSet());
+    if (collisions.size()) {
+        for (const QString &name : collisions) {
+            // Add number to device model names created by user
+            MerEmulatorDeviceModel model = userDeviceModels.take(name);
+
+            int num = 0;
+            while (userDeviceModels.contains(name + QString::number(num)))
+                ++num;
+
+            model.setName(name + QString::number(num));
```

Comment:
Duplicated code. I would prefer encapsulating it in a function `QString uniqueName(QString baseName, QSet<QString> existingNames)` (make it a static member of MerDeviceModel?). I would also consider using `" (<num>)"` suffix. Isn't there such a function in Utils already?

Reply:
Fixed. I have to add MerEmulatorDeviceModel::uniqueName function, because most relevant function Project::makeUnique is not fit.